From 9817b04740206c0640d3ae9ea9c30f777e26389b Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Fri, 24 Jun 2022 11:23:20 +0300
Subject: [PATCH 15/33] pci: s32cc: Remove the usage of 'device_id' property

The id of PCIe device can be obtained based on node alias.

Issue: ALB-8357
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/pci/pcie_s32gen1.c   | 6 +++++-
 drivers/pci/serdes_s32gen1.c | 7 ++++++-
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/pcie_s32gen1.c b/drivers/pci/pcie_s32gen1.c
index 0b92c7754b..b361862d64 100644
--- a/drivers/pci/pcie_s32gen1.c
+++ b/drivers/pci/pcie_s32gen1.c
@@ -717,7 +717,11 @@ static int s32_pcie_get_config_from_device_tree(struct s32_pcie *pcie)
 
 	debug("%s: dt node: %d\n", __func__, node);
 
-	pcie->id = fdtdec_get_int(fdt, node, "device_id", -1);
+	ret = dev_read_alias_seq(pcie->bus, &pcie->id);
+	if (ret) {
+		printf("Failed to get PCIe device id\n");
+		return ret;
+	}
 
 	ret = fdt_get_named_resource(fdt, node, "reg", "reg-names",
 				     "dbi", &pcie->dbi_res);
diff --git a/drivers/pci/serdes_s32gen1.c b/drivers/pci/serdes_s32gen1.c
index eb7cfb36f1..ba8c222ca7 100644
--- a/drivers/pci/serdes_s32gen1.c
+++ b/drivers/pci/serdes_s32gen1.c
@@ -758,7 +758,12 @@ static int s32_serdes_get_config_from_device_tree(struct s32_serdes *pcie)
 	debug("%s: dbi: 0x%p (0x%p)\n", __func__, (void *)pcie->dbi_res.start,
 	      pcie->dbi);
 
-	pcie->id = fdtdec_get_int(fdt, node, "device_id", -1);
+	ret = fdtdec_get_alias_seq(gd->fdt_blob, "serdes",
+				   node, &pcie->id);
+	if (ret) {
+		printf("Failed to get SerDes device id\n");
+		return ret;
+	}
 
 	/* get supported width (X1/X2) from device tree */
 	pcie->linkwidth = fdtdec_get_int(fdt, node, "num-lanes", X1);
-- 
2.17.1

